packages/ui/src/machines/authMachine.ts

```diff
@@ -17,7 +17,9 @@ export const authMachine = createMachine<AuthContext, AuthEvent>(
     initial: 'idle',
     context: {
       user: undefined,
-      config: undefined,
+      config: {
+        login_mechanisms: ['username'],
```

Comment:
Defaulted in the state machine so that all view code can pass along their `loginMechanisms`, even if undefined.

**Note – I'm only keeping `config.login_mechanisms` because of how many references there are to it!  Ideally, all `context.*` keys are _flat_, so we don't have to worry about deep-merging arrays**.